File: widget/src/main/java/com/ooftf/widget/activity/NotificationActivity.java
package com.ooftf.widget.activity;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.graphics.BitmapFactory;
import android.os.Build;
import android.os.Bundle;
import android.os.SystemClock;
import android.support.v4.app.NotificationCompat;
import android.support.v4.app.NotificationManagerCompat;
import android.widget.Button;

import com.alibaba.android.arouter.facade.annotation.Route;
import com.ooftf.service.base.BaseActivity;
import com.ooftf.service.utils.JLog;
import com.ooftf.widget.R;
import com.ooftf.widget.R2;
import com.ooftf.widget.notification.NotificationChannelManager;
import com.ooftf.widget.widget.NewMessageNotification;

import butterknife.BindView;
import butterknife.ButterKnife;

import static android.os.Build.VERSION.SDK_INT;
import static android.os.Build.VERSION_CODES.JELLY_BEAN;
import static android.os.Build.VERSION_CODES.O;

/**
 * @author ooftf
 */
@Route(path = "/widget/activity/notification")
public class NotificationActivity extends BaseActivity {
    private static final String NOTIFICATION_CHANNEL_ID = "leakcanary";
    @BindView(R2.id.show)
    Button show;
    @BindView(R2.id.show2)
    Button show2;
    @BindView(R2.id.show3)
    Button show3;
    NotificationManager notificationManager;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_notification);
        ButterKnife.bind(this);
        notificationManager =  ((NotificationManager) this.getSystemService(Context.NOTIFICATION_SERVICE));
        show.setOnClickListener(v -> {
            showNotification();
        });
        show2.setOnClickListener(v -> {
            NewMessageNotification.notify(NotificationActivity.this, "这是什么啊", 15);
        });
        show3.setOnClickListener(v -> {
            JLog.e(NotificationChannelManager.getChannel(this, "One"));
        });
    }

    private void showNotificationCompat() {
        NotificationCompat.Builder builder = new NotificationCompat.Builder(this, "One");
        Notification notification = builder
                .setContentText("setContentText")
                .setContentTitle("setContentTitle")
                .setSmallIcon(R.mipmap.ic_stat_new_message)
                .setLargeIcon(BitmapFactory.decodeResource(getResources(), R.mipmap.example_picture))
                .build();
        NotificationManagerCompat.from(this).notify(55, notification);
    }

    private void showNotification() {
        NotificationManager notificationManager = ((NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE));
        NotificationCompat.Builder builder;
        builder = new NotificationCompat.Builder(this, NotificationChannelManager.getOneChannel(this));
        builder.setContentTitle("setContentTitle")
                .setContentText("setContentText")
                .setLargeIcon(BitmapFactory.decodeResource(getResources(), R.mipmap.example_picture))
                .setTicker("this is ticker")
                .setWhen(System.currentTimeMillis())
                .setOnlyAlertOnce(true)
                .setSmallIcon(R.mipmap.ic_stat_new_message);


        Notification notification = builder
                .build();
        notificationManager.notify(55, notification);

    }


}

File: m-widget/src/main/java/com/ooftf/widget/activity/NotificationActivity.kt
package com.ooftf.widget.activity

import android.app.NotificationManager
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.View
import androidx.core.app.NotificationCompat
import androidx.core.app.NotificationManagerCompat
import com.alibaba.android.arouter.facade.annotation.Route
import com.ooftf.arch.frame.mvvm.activity.BaseViewBindingActivity
import com.ooftf.log.JLog
import com.ooftf.widget.R
import com.ooftf.widget.databinding.ActivityNotificationBinding
import com.ooftf.widget.notification.NotificationChannelManager
import com.ooftf.widget.widget.NewMessageNotification

/**
 * @author ooftf
 */
@Route(path = "/widget/activity/notification")
class NotificationActivity : BaseViewBindingActivity<ActivityNotificationBinding>() {
    var notificationManager: NotificationManager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        notificationManager = this.getSystemService(NOTIFICATION_SERVICE) as NotificationManager
        binding.show.setOnClickListener { v: View? -> showNotification() }
        binding.show2.setOnClickListener { v: View? -> NewMessageNotification.notify(this@NotificationActivity, "这是什么啊", 15) }
        binding.show3.setOnClickListener { v: View? -> JLog.e(NotificationChannelManager.getChannel(this, "One")) }
    }

    private fun showNotificationCompat() {
        val builder = NotificationCompat.Builder(this, "One")
        val notification = builder
                .setContentText("setContentText")
                .setContentTitle("setContentTitle")
                .setSmallIcon(R.mipmap.ic_stat_new_message)
                .setLargeIcon(BitmapFactory.decodeResource(resources, R.mipmap.example_picture))
                .build()
        NotificationManagerCompat.from(this).notify(55, notification)
    }

    private fun showNotification() {
        val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
        val builder: NotificationCompat.Builder
        builder = NotificationCompat.Builder(this, NotificationChannelManager.getOneChannel(this))
        builder.setContentTitle("setContentTitle")
                .setContentText("setContentText")
                .setLargeIcon(BitmapFactory.decodeResource(resources, R.mipmap.example_picture))
                .setTicker("this is ticker")
                .setWhen(System.currentTimeMillis())
                .setOnlyAlertOnce(true)
                .setSmallIcon(R.mipmap.ic_stat_new_message)
        val notification = builder
                .build()
        notificationManager.notify(55, notification)
    }

    companion object {
        private const val NOTIFICATION_CHANNEL_ID = "leakcanary"
    }
}